homeassistant/components/ezviz/siren.py

```diff
@@ -0,0 +1,114 @@
+"""Support for EZVIZ sirens."""
+from __future__ import annotations
+
+from datetime import datetime, timedelta
+from typing import Any
+
+from pyezviz import HTTPError, PyEzvizError, SupportExt
+
+from homeassistant.components.siren import (
+    SirenEntity,
+    SirenEntityDescription,
+    SirenEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_ON
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+import homeassistant.helpers.event as evt
+from homeassistant.helpers.restore_state import RestoreEntity
+
+from .const import DATA_COORDINATOR, DOMAIN
+from .coordinator import EzvizDataUpdateCoordinator
+from .entity import EzvizEntity
+
+PARALLEL_UPDATES = 1
+OFF_DELAY = timedelta(seconds=60)  # Camera firmware has hard coded turn off.
+
+SIREN_ENTITY_TYPE = SirenEntityDescription(
+    key="siren",
+    translation_key="siren",
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up EZVIZ sensors based on a config entry."""
+    coordinator: EzvizDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id][
+        DATA_COORDINATOR
+    ]
+
+    async_add_entities(
+        EzvizSirenEntity(coordinator, camera)
+        for camera in coordinator.data
+        for capibility, value in coordinator.data[camera]["supportExt"].items()
+        if capibility == str(SupportExt.SupportActiveDefense.value)
+        if value != "0"
+    )
+
+
+class EzvizSirenEntity(EzvizEntity, SirenEntity, RestoreEntity):
+    """Representation of a EZVIZ Siren entity."""
+
+    _attr_has_entity_name = True
+    _attr_supported_features = SirenEntityFeature.TURN_ON | SirenEntityFeature.TURN_OFF
+
+    def __init__(
+        self,
+        coordinator: EzvizDataUpdateCoordinator,
+        serial: str,
+    ) -> None:
+        """Initialize the Siren."""
+        super().__init__(coordinator, serial)
+        self._attr_unique_id = f"{serial}_{SIREN_ENTITY_TYPE.key}"
```

Comment:
Instead of `SIREN_ENTITY_TYPE.key` use `description.key` (pass in description entity) or skip entity description all together as I suppose no more entities will be added here